From 9098b56b72f674d5c6a278ef3aaf0df2fe247262 Mon Sep 17 00:00:00 2001
From: "Wang, Yu" <yu.y.wang@intel.com>
Date: Thu, 15 Aug 2013 18:39:36 +0800
Subject: [PATCH 222/429] dwc-intel: Fixes incorrect USB charger capability.

SMIP set:
1, USB2 SDP, only send 500mA. Ignore suspend/resume.
2, USB3 SDP, only send 900mA. Ignore suspend/resume.
3, D+/D-, only send 500mA.

SMIP not set:
1, USB2 SDP, 100mA -> 500mA -> 0mA(suspend) -> 500mA(resume).
2, USB3 SDP, 150mA -> 900mA -> 0mA(suspend) -> 900mA(resume).
3, D+/D- only send 0mA.

Signed-off-by: Wang, Yu <yu.y.wang@intel.com>
---
 drivers/usb/dwc3/otg.c              | 7 ++-----
 drivers/usb/dwc3/otg.h              | 7 +++++++
 include/linux/usb/dwc3-intel-mrfl.h | 1 +
 3 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/drivers/usb/dwc3/otg.c b/drivers/usb/dwc3/otg.c
index d76f478e321f..65ac52207b07 100644
--- a/drivers/usb/dwc3/otg.c
+++ b/drivers/usb/dwc3/otg.c
@@ -458,10 +458,6 @@ static enum dwc_otg_state do_charger_detection(struct dwc_otg2 *otg)
 	case POWER_SUPPLY_CHARGER_TYPE_SE1:
 		ma = 1500;
 		break;
-	case POWER_SUPPLY_CHARGER_TYPE_USB_SDP:
-		/* Notify SDP current is 100ma before enumeration. */
-		ma = 100;
-		break;
 	default:
 		otg_err(otg, "Charger type is not valid to notify battery\n");
 		return -EINVAL;
@@ -476,12 +472,13 @@ static enum dwc_otg_state do_charger_detection(struct dwc_otg2 *otg)
 	case POWER_SUPPLY_CHARGER_TYPE_ACA_DOCK:
 	case POWER_SUPPLY_CHARGER_TYPE_USB_DCP:
 	case POWER_SUPPLY_CHARGER_TYPE_USB_CDP:
-	case POWER_SUPPLY_CHARGER_TYPE_USB_SDP:
 	case POWER_SUPPLY_CHARGER_TYPE_SE1:
 		if (dwc_otg_notify_charger_type(otg,
 					POWER_SUPPLY_CHARGER_EVENT_CONNECT) < 0)
 			otg_err(otg, "Notify battery type failed!\n");
 		break;
+	case POWER_SUPPLY_CHARGER_TYPE_USB_SDP:
+	/* SDP is complicate, it will be handle in set_power */
 	default:
 		break;
 	}
diff --git a/drivers/usb/dwc3/otg.h b/drivers/usb/dwc3/otg.h
index 6032c43567a8..a5df0050241f 100644
--- a/drivers/usb/dwc3/otg.h
+++ b/drivers/usb/dwc3/otg.h
@@ -414,6 +414,13 @@ struct dwc3_otg_hw_ops {
 	int (*resume)(struct dwc_otg2 *otg);
 };
 
+#define OTG_USB2_100MA				0xfff1
+#define OTG_USB3_150MA				0xfff2
+#define OTG_USB2_500MA				0xfff3
+#define OTG_USB3_900MA				0xfff4
+#define OTG_DEVICE_SUSPEND			0xfffe
+#define OTG_DEVICE_RESUME			0xffff
+
 void dwc3_wakeup_otg_thread(struct dwc_otg2 *otg);
 struct dwc_otg2 *dwc3_get_otg(void);
 int dwc3_otg_register(struct dwc3_otg_hw_ops *pdata);
diff --git a/include/linux/usb/dwc3-intel-mrfl.h b/include/linux/usb/dwc3-intel-mrfl.h
index f8168313cf98..82f30995a611 100644
--- a/include/linux/usb/dwc3-intel-mrfl.h
+++ b/include/linux/usb/dwc3-intel-mrfl.h
@@ -24,6 +24,7 @@
 
 struct intel_dwc_otg_pdata {
 	int is_hvp;
+	int charging_compliance;
 };
 
 #define TUSB1211_VENDOR_ID_LO					0x00
-- 
2.37.3

